.github/workflows/build-and-publish.yml

```diff
@@ -24,10 +24,26 @@ on:
         type: string
 
 jobs:
+  get-commit-hash:
+    name: Get commit hash
+    runs-on: ubuntu-latest
+    outputs:
+      commit_hash: ${{ steps.get-commit-hash.outputs.commit_hash }}
```

Comment:
I assume you've tested this? I've never seen this syntax before

Reply:
take a look at the testing notes in the description and let me know if you think i missed something